query/physicalplan/aggregate.go

```diff
@@ -281,18 +282,20 @@ func hashArray(arr arrow.Array) []uint64 {
 
 func hashListArray(arr *array.List) []uint64 {
 	res := make([]uint64, arr.Len())
+	digest := xxhash.New()
 	for i := 0; i < arr.Len(); i++ {
-		list := []byte{}
 		if err := arrowutils.ForEachValueInList(i, arr, func(_ int, v any) {
 			switch val := v.(type) {
 			case []byte:
-				list = append(list, val...)
+				// NOTE: We can ignore the return here because from the xxhash Write function is says: It always returns len(b), nil.
+				_, _ = digest.Write(val)
```

Comment:
Just double checking: would e.g. `[a, b, c]` and `[b, c, a]` hash to different values?

Reply:
Correct; can see here: https://go.dev/play/p/rKO6WL2JK54